resip/stack/TransportSelector.cxx

```diff
@@ -1547,15 +1547,14 @@ TransportSelector::findTransportBySource(Tuple& search, const SipMessage* msg) c
 {
    DebugLog(<< "findTransportBySource(" << search << ")");
 
-   if(msg && 
-      !msg->getTlsDomain().empty() && 
-      isSecure(search.getType()))
+   if(msg && isSecure(search.getType()))
    {
       // We should not be willing to attempt sending on a TLS/DTLS/WSS transport 
```

Comment:
I assume adding a TLS transport without a domain is only valid to do if resip is going to as a TLS client only, and will not accept inbound TLS connections.  Is that correct?

Assuming so, this change makes the comment text invalid, and it should be updated.

Reply:
Actually, we are talking about the transport selection which is used by TransportSelector::transmit(). It means an outgoing SIP message sending but not TLS handshake.
That's why this code can be applied to a service TLS listening transports.

Let's imagine Registrar listens to `TLS:1.2.3.4:5061` where no domain is assigned.
The service has the functionality to send SIP requests, like SIP OPTIONS for NAT pings and UA is registered via TLS. 
Of course, there must be an SSL certificate with the particular IP address to create a TLS connection;
however, transport selection seems used to find which interface the service is going to create/re-use the connection.

To be honest, I do not understand totally the comment "We should not be willing ... then proceed with the code below.". I expect that a service chooses the transport for sending SIP requests and only after that TLS connection is created/re-used. It looks like two separate layers. Correct me if I am wrong.

On the other hand, I removed the condition `!msg->getTlsDomain().empty()` as a TLS transport is added in a separate container and it doesn't work for such selections, see  https://github.com/resiprocate/resiprocate/blob/master/resip/stack/TransportSelector.cxx#L253

Reply:
In addition, if to talk about "a TLS transport without a domain", I believe it is not so popular to bind TLS transport without a domain on the Internet. However, it is not forbidden.
For example, you have a small office with a SIP server installation and don't want to purchase a domain. It should work well.

Reply:
I think the comment is saying that if the sipMessage has a TlsDomain set on it, then we should only choose the TLS transport for that domain to send on (ie: the transport that has that domain cert set on it that matches the domain set in the SipMessage).   

That being said I'm not sure offhand how, when or why the TlsDomain get's set on a SipMessage.